CONTRIBUTING.md

```diff
@@ -0,0 +1,85 @@
+
+### Opening an Issue
+
+If you think you have found a bug, or have a new feature idea, please start by making sure it hasn't already been [reported or fixed](https://github.com/callemall/material-ui/issues?utf8=%E2%9C%93&q=is%3Aopen+is%3Aclosed). You can search through existing issues and PRs to see if someone has reported one similar to yours.
+
+Next, create a new issue that briefly explains the problem, and provides a bit of background as to the circumstances that triggered it, and steps to reproduce it.
+
+For code issues please include:
+* Material UI version
+* React version
+* Browser version
+* A code example or link to a repo, gist or running site.
+
+For visual or layout problems, images or animated gifs can help explain your issue.
+
+For feature requests please include a link to the relevant section of Material Design spec, or a screenshot.
+
+## Issue Guidelines
+
+Please begin the title with '[ComponentName]' where appropriate, and use a succint description. "doesn't work" doesn't help others find similar issues.
+
+Please don't group multiple topics into one issue, but instead each should be its own issue.
+
+And please don't just '+1' an issue. It spams the maintainers and doesn't help move the issue forward.
+
+### Submitting a Pull Request
+
+Material-UI is a community project, so pull requests are always welcome, but before working on a large change, it is best to open an issue first to discuss it with the maintainers.
+
+When in doubt, keep your pull requests small. To give a PR the best chance of getting accepted, do not bundle more than one feature or bug fix per pull request. It's always best to create two smaller PRs than one big one.
+
+As with issues, please begin the title with [ComponentName].
+
+### Getting started
+
+Please create a new branch from an up to date master on your fork.
+
+1. Fork the Material UI repository on Github
+2. Clone your fork to your local machine `git clone git@github.com:<yourname>/material-ui.git`
+3. Create a branch `git checkout -b my-topic-branch`
+4. Make your changes, lint, then push to to github with `git push --set-upstream origin my-topic-branch`.
+5. Visit github and make your pull request.
+
+If you have an existing local repositiry, please update it before you start to minimise the chance of merge conflicts.
+```js
+git remote add upstream git@github.com:callemall/material-ui.git
+git checkout master
+git pull upstream master
+git checkout -b my-topic-branch
+npm update
+```
+
+### Testing
+
+The documentation site is built with Material UI, and contains examples of all the components. To get started:
+```js
+npm install
+cd docs
+npm install
+npm run start
+```
+You can now access
+If you find you need to use `console.log()` or `debugger`, you will need to temporarily disable linting. Simply comment out the [`preLoaders` section](https://github.com/callemall/material-ui/blob/master/docs/webpack-dev-server.config.js#L62) of `docs/webpack-dev-server.config.js`
```

Comment:
This was a *pain in the ass* for me. We should find a way to make any eslint error not blocking the buidling.